src/utils/getProductImage.ts

```diff
@@ -107,4 +108,50 @@ const generateOptimizedImages = (
   return imageUrlArray;
 };
 
-export { generateOptimizedImages, getProductImageURLs };
+function getProductImagesFromQuery(item: Product) {
+  const { productView } = item;
+  const attributeId = productView?.options?.[0].id;
+  if (!attributeId) {
+    return [];
+  }
+
+  const imageAttributes = productView?.attributes?.find(({ name }) => name === 'eds_images');
```

Comment:
Would be great if you can add a comment explaining all this crazy logic to select the right image.

Reply:
Added comment